packages/xo-web/src/xo-app/xostor/new-xostor-form.js

```diff
@@ -0,0 +1,542 @@
+        }
+      },
+      poolHasGoodNumberOfHosts: state => hasGoodNumberOfHosts(state.poolHosts),
+      poolHasAXostor: (state, props) => hasAXostor(props.srs[state.poolId]),
+      isPoolXostorCompatible: state => state.isXcpngHost && state.poolHasGoodNumberOfHosts && !state.poolHasAXostor,
+    },
+  }),
+  injectState,
+  ({ effects, state }) => (
+    <Card>
+      <CardHeader>{_('pool')}</CardHeader>
+      <CardBlock>
+        <div>
+          <label>
+            <input
+              checked={state.onlyShowXostorPools}
+              name='onlyShowXostorPools'
+              onChange={effects.toggleState}
+              type='checkbox'
+            />{' '}
+            {_('onlyShowXostorRequirements', { type: _('pools') })}
+          </label>
+          <SelectPool onChange={effects.onPoolChange} predicate={state.poolPredicate} value={state.poolId} />
+          {state.poolHosts !== undefined && !state.isPoolXostorCompatible && (
+            <div className='text-danger'>
+              {/* FIXME: add link of the documentation when ready */}
+              <a href='#' rel='noreferrer' target='_blank'>
+                {_('objectDoesNotMeetXostorRequirements', { object: <PoolRenderItem id={state.poolId} /> })}
+              </a>
+              <ul>
+                {!state.isXcpngHost && <li>{_('notXcpPool')}</li>}
+                {!state.poolHasGoodNumberOfHosts && <li>{_('wrongNumberOfHosts')}</li>}
+                {state.poolHasAXostor && <li>{_('poolAlreadyHasXostor')}</li>}
+              </ul>
+            </div>
+          )}
+          <em>
+            <Icon icon='info' /> {_('xostorPackagesWillBeInstalled')}
+          </em>
+        </div>
+      </CardBlock>
+    </Card>
+  ),
+])
+
+const NetworkCard = decorate([
+  provideState({
+    initialState: () => ({ onlyShowXostorNetworks: true }),
+    effects: {
+      toggleState,
+    },
+    computed: {
+      networksPredicate: (state, props) => network => {
+        const isOnPool = network.$pool === state.poolId
+        const pifs = network.PIFs
+        return state.onlyShowXostorNetworks
+          ? isOnPool && pifs.length > 0 && pifs.every(pifId => props.pifs[pifId].ip !== '')
+          : isOnPool
+      },
+    },
+  }),
+  injectState,
+  ({ effects, state }) => (
+    <Card>
+      <CardHeader>{_('network')}</CardHeader>
+      <CardBlock>
+        <label>
+          <input
+            checked={state.onlyShowXostorNetworks}
+            name='onlyShowXostorNetworks'
+            onChange={effects.toggleState}
+            type='checkbox'
+          />{' '}
+          {_('onlyShowXostorRequirements', { type: _('networks') })}
+        </label>
+        <SelectNetwork
+          disabled={!state.isPoolSelected}
+          onChange={effects.onNetworkChange}
+          predicate={state.networksPredicate}
+          value={state.networkId}
+        />
+      </CardBlock>
+    </Card>
+  ),
+])
+
+const DisksCard = decorate([
```

Comment:
I find it a bit strange that to add disks to each host, you have to re-select a new host and re-select the disk you want for that host, and so on. Can you discuss this with Clémence?